org-eclipse-che-jdt-ls-extension/jdt.ls.extension.api/src/main/java/org/eclipse/che/jdt/ls/extension/api/dto/ResourceLocationParameters.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.api.dto;
+
+/**
+ * Resource location parameters.
+ *
+ * @author Anatolii Bazko
+ */
+public class ResourceLocationParameters {
```

Comment:
I think this should really be a Either<String, ResourceLocation> where ResourceLocation is <fqn, libid>

Reply:
Clarify pls.

Reply:
This class is really two classes (either it has fqn, libid, or a file uri, but never any other combination). The correct way to express this in lsp4j is using Either<L, R>, not dispatch based on field values.

Reply:
Should I use `Either<L, R>` as returned value in command?